### DISCUSSION TITLE
Add confirmation when automatically forking a repo for use with the built-in editor

### DISCUSSION THREAD

**Discussion by micahswitzer** [community/community#16729] [Other Feature Feedback, Questions, & Ideas]
Currently if you press `e` while viewing a file from a repo for which you do not have push permissions, GitHub will automatically, without prompt, create a fork of that repo for you to edit and commit changes to that file. If that press of `e` was accidental and you want to delete the fork, you then have to go your fork's settings and go through the (rightfully) tedious process of 1. re-authenticating, and 2. typing the fork's name to delete it. All because you bumped `e` while browsing some source code.

I propose that a simple confirmation dialog be shown in this situation stating what will happen and allowing the user to quickly cancel if that's not what they meant/wanted.

This has happened to me more than once and was incredibly frustrating both times which is why I'm now creating this discussion.

**Comment by JohnVeness** [comment]
I came here to suggest similar. However, I think a nicer user experience than a prompt would be that if you clicked to edit accidentally then cancelled (e.g. pressed Back), without actually making any changes, the fork is deleted automatically.

Or indeed maybe the fork isn't actually created until you save any edits you make.

**Comment by jdimatteo** [comment]
This is dangerous for private repos, because forks are public by default, so users could easily accidentally leak private repos!
